### PR TITLE
DOC: Clarify digital filter functions

### PR DESCRIPTION
**Pull request by endolith**
From https://dsp.stackexchange.com/a/42403/29 (trying to use filtfilt
with analog filter coefficients)

Also write "A one-line summary that does not use variable names or
the function name"  `lfiltic` vs `lfilter_zi` has always been
confusing to me.

Move filtfilt content to Notes section

### CODE REVIEW

**Comment by endolith** [comment]
Also maybe should have an example like this?

https://dsp.stackexchange.com/a/19086/29

File: scipy/signal/signaltools.py [hunk 1]
```diff
@@ -2916,26 +2920,14 @@ def _filtfilt_gust(b, a, x, axis=-1, irlen=None):
 def filtfilt(b, a, x, axis=-1, padtype='odd', padlen=None, method='pad',
              irlen=None):
     """
-    A forward-backward filter.
+    Apply a digital filter forward and backward to a signal
```
Comment (Eric89GXL) [inline scipy/signal/signaltools.py]: Might as well add the trailing `.` here

**Review by larsoner — APPROVED** [review]
LGTM even without my nitpick.

I agree it would be nice to have the filtfilt / lfilter example, but can be another PR if you want.

**Comment by endolith** [comment]
Ok fixed those.  The example is kind of long though and this function already has lots of examples

**Comment by Eric89GXL** [comment]
Thanks @endolith

**Pull request merged**